src/basic/terminal-util.c

```diff
@@ -1098,13 +1098,9 @@ int get_ctty(pid_t pid, dev_t *ret_devnr, char **ret) {
                 return -EINVAL;
 
         if (ret) {
-                _cleanup_free_ char *b = NULL;
-
-                b = strdup(w);
-                if (!b)
-                        return -ENOMEM;
-
-                *ret = TAKE_PTR(b);
+                r = strdup_to(ret, w);
+                if (r < 0)
+                        return r;
         }
```

Comment:
`strdup_to()` asserts on the first pointer. The reasoning is that in virtually all cases, it's called with a non-NULL param, and since it always returns 0, it can only return -ENOMEM, so I added the assert.

Reply:
oh, uh, weird. strdup_to() and strdup_to_full() behaving differently in that regard is really strange.

I'd just drop the ASSERT_PTR() on that tbh, seems like an unnecessary pitfall. Nothing is really won.